Formula/archey4.rb

```diff
@@ -0,0 +1,29 @@
+class Archey4 < Formula
+  include Language::Python::Virtualenv
+
+  desc "Simple system information tool written in Python"
+  homepage "https://github.com/HorlogeSkynet/archey4"
+  url "https://files.pythonhosted.org/packages/68/7d/8066c39289bda86c606350aca46b05e157eed2b6391af75b5745cbd6f61c/archey4-4.11.0.tar.gz"
+  sha256 "18113b18282ea0ab31acd74d22d2830160ddf0efc82438e328b3ca9823e15666"
+  license "GPL-3.0-or-later"
+
+  depends_on "python@3.9"
+
+  resource "distro" do
+    url "https://files.pythonhosted.org/packages/a6/a4/75064c334d8ae433445a20816b788700db1651f21bdb0af33db2aab142fe/distro-1.5.0.tar.gz"
+    sha256 "0e58756ae38fbd8fc3020d54badb8eae17c5b9dcbed388b17bb55b8a5928df92"
+  end
+
+  resource "netifaces" do
+    url "https://files.pythonhosted.org/packages/0d/18/fd6e9c71a35b67a73160ec80a49da63d1eed2d2055054cc2995714949132/netifaces-0.10.9.tar.gz"
+    sha256 "2dee9ffdd16292878336a58d04a20f0ffe95555465fee7c9bd23b3490ef2abf3"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/archey", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for your quick feedback @SMillerDev.
IMHO this is a _special_ case as we can't easily test one of the functionalities here.

There is no way I can guarantee you :

* a specific output (everything is system-dependent)
* a file would be created (only console output)

That's why I've just replaced `-v` flag by `-j`, which now implies a simple execution with minimal (relevant) output we can expect from it. If returned status code is zero, everything should be fine.

Bye :wave:

Reply:
> a specific output (everything is system-dependent)

I'm not sure what information it provides, but you could check for things like "macOS|linux", "cpu", "directory" since those are gonna be possible in all hardware configurations.

Reply:
Please find a new patch implementing your proposal adapted for current supported platforms @SMillerDev.